packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerCloudState.ts

```diff
@@ -103,6 +103,11 @@ export class LayerCloudState {
         // eslint-disable-next-line no-param-reassign
         layerVersion.legacyLayer = layerVersion.LogicalName === undefined || layerVersion.LogicalName === 'LambdaLayer';
       });
+      if (layerVersionList.length === 0) {
+        throw new AmplifyError('LambdaLayerNotFoundError', {
+          message: 'No versions were found for the Lambda Layer. Were they deleted on the AWS Lambda Console?',
+        });
+      }
```

Comment:
Should we do this before line 45 in this file? Maybe also add the layer name and environment in the error message?

Reply:
Good suggestions. The error message now includes the name of the missing layers.